src/coreclr/debug/shared/dbgtransportsession.cpp

```diff
@@ -668,16 +675,14 @@ HRESULT DbgTransportSession::SendMessage(Message *pMessage, bool fWaitsForReply)
             pMessage = pMessageCopy;
         }
 
-        // Check the session state.
-        if (m_eState == SS_Closed)
+        // If the state is SS_Open we can send the message now.
+        if (m_eState == SS_Open)
```

Comment:
This send was moved before the queuing of the message. I assume this is not a problem because we're holding a state lock.